cranelift/interpreter/src/value.rs

```diff
@@ -38,15 +41,21 @@ pub trait DataValueExt: Sized {
     fn add(self, other: Self) -> ValueResult<Self>;
     fn sub(self, other: Self) -> ValueResult<Self>;
     fn mul(self, other: Self) -> ValueResult<Self>;
-    fn div(self, other: Self) -> ValueResult<Self>;
-    fn rem(self, other: Self) -> ValueResult<Self>;
+    fn udiv(self, other: Self) -> ValueResult<Self>;
+    fn sdiv(self, other: Self) -> ValueResult<Self>;
+    fn urem(self, other: Self) -> ValueResult<Self>;
+    fn srem(self, other: Self) -> ValueResult<Self>;
     fn sqrt(self) -> ValueResult<Self>;
     fn fma(self, a: Self, b: Self) -> ValueResult<Self>;
     fn abs(self) -> ValueResult<Self>;
-    fn checked_add(self, other: Self) -> ValueResult<Option<Self>>;
-    fn overflowing_add(self, other: Self) -> ValueResult<(Self, bool)>;
-    fn overflowing_sub(self, other: Self) -> ValueResult<(Self, bool)>;
-    fn overflowing_mul(self, other: Self) -> ValueResult<(Self, bool)>;
+    fn uchecked_add(self, other: Self) -> ValueResult<Option<Self>>;
+    fn schecked_add(self, other: Self) -> ValueResult<Option<Self>>;
+    fn uoverflowing_add(self, other: Self) -> ValueResult<(Self, bool)>;
+    fn soverflowing_add(self, other: Self) -> ValueResult<(Self, bool)>;
+    fn uoverflowing_sub(self, other: Self) -> ValueResult<(Self, bool)>;
+    fn soverflowing_sub(self, other: Self) -> ValueResult<(Self, bool)>;
+    fn uoverflowing_mul(self, other: Self) -> ValueResult<(Self, bool)>;
+    fn soverflowing_mul(self, other: Self) -> ValueResult<(Self, bool)>;
```

Comment:
Small nit about these. If we're following the IR naming scheme I think these would be better named as `sadd_overflow`/`smul_overflow`/etc.. since that is the name of the opcode.

It's also really neat that the other operations in `DataValue` now have the names of the opcodes!

Reply:
I changed it. Hope it's better now